### PR TITLE
Adds patch info to OE Rel notes

### PR DESCRIPTION
**Pull request by lamagnifica**

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
## [DOC-2767](https://openedx.atlassian.net/browse/DOC-2767)

We updated the Dogwood version of the Installation guide to point people ONLY to the wiki for git tag and box name values. This brings the same change into the master/latest version.
This also adds a patch release section to the Dogwood release notes topic.
There was an error in the build, so the final two commits fix a link. 

### Date Needed (optional)

ASAFeasible

### Reviewers
- [ ] Subject matter expert: @nedbat 
- [x] Doc team review (sanity check): @pdesjardins @srpearce @catong 

### Testing
- [x] Ran ./run_tests.sh without warnings or errors

### Post-review
- [ ] Add description to release notes task as a comment
- [ ] Squash commits

**Comment by nedbat** [comment]
@lamagnifica btw, it looks like if you tag someone by editing a comment, there's no notification sent... :(  Not sure where this leaves us with the cover letters...

File: en_us/open_edx_release_notes/source/dogwood.rst [hunk 1]
```diff
@@ -487,6 +487,26 @@ sent to your email account.
...
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Date
+     - Change
+   * - 9 March 2016
+     - An update to the release packaging was made. To verify that you have the
+       most up-to-date Git tag and Vagrant box name for this release, see the
```
Comment (nedbat) [inline en_us/open_edx_release_notes/source/dogwood.rst]: This exhortation to check for an up-to-date tag will have people looking for update instructions, and trying to update, but they don't need to.

File: en_us/open_edx_release_notes/source/dogwood.rst [hunk 2]
```diff
@@ -487,6 +487,26 @@ sent to your email account.
...
+       `Open edX Releases Wiki page`_.
+   * -
+     - The help links in Studio now open topics in the correct version of the
+       *Building and Running an Open edX* Course guide.
```
Comment (srpearce) [inline en_us/open_edx_release_notes/source/dogwood.rst]: Course is outside the italic asterisk here

**Comment by srpearce** [comment]
One quick typo fix; otherwise :+1:

File: en_us/open_edx_release_notes/source/dogwood.rst [hunk 3]
```diff
@@ -487,6 +487,24 @@ sent to your email account.
...
+**************
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Date
+     - Change
+   * - 9 March 2016
+     - An update to the release packaging was made.
```
Comment (lamagnifica) [inline en_us/open_edx_release_notes/source/dogwood.rst:501]: @nedbat less is more? ^^

Comment (nedbat) [inline en_us/open_edx_release_notes/source/dogwood.rst:501]: Beautifully minimal! :)

**Pull request merged**